benches/read_metadata.rs

```diff
@@ -1,38 +1,129 @@
 use bencher::{benchmark_group, benchmark_main};
 
-use std::io::{Cursor, Write};
+use std::fs;
+use std::io::{prelude::*, Cursor};
 
 use bencher::Bencher;
+use getrandom::getrandom;
+use tempdir::TempDir;
 use zip::write::SimpleFileOptions;
-use zip::{CompressionMethod, ZipArchive, ZipWriter};
+use zip::{result::ZipResult, CompressionMethod, ZipArchive, ZipWriter};
 
 const FILE_COUNT: usize = 15_000;
 const FILE_SIZE: usize = 1024;
 
-fn generate_random_archive(count_files: usize, file_size: usize) -> Vec<u8> {
+fn generate_random_archive(count_files: usize, file_size: usize) -> ZipResult<Vec<u8>> {
     let data = Vec::new();
     let mut writer = ZipWriter::new(Cursor::new(data));
     let options = SimpleFileOptions::default().compression_method(CompressionMethod::Stored);
 
-    let bytes = vec![0u8; file_size];
+    let mut bytes = vec![0u8; file_size];
 
     for i in 0..count_files {
         let name = format!("file_deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_{i}.dat");
-        writer.start_file(name, options).unwrap();
-        writer.write_all(&bytes).unwrap();
+        writer.start_file(name, options)?;
+        getrandom(&mut bytes).unwrap();
+        writer.write_all(&bytes)?;
     }
 
-    writer.finish().unwrap().into_inner()
+    Ok(writer.finish()?.into_inner())
 }
 
 fn read_metadata(bench: &mut Bencher) {
-    let bytes = generate_random_archive(FILE_COUNT, FILE_SIZE);
+    let bytes = generate_random_archive(FILE_COUNT, FILE_SIZE).unwrap();
```

Comment:
So we need to make `read_metadata() -> ZipResult<()>` to do this, but when I try that, I get this:
```rust
    = note: expected fn pointer `for<'a> fn(&'a mut Bencher)`
                  found fn item `for<'a> fn(&'a mut Bencher) -> Result<(), ZipError> {read_metadata}`
``` 

I have used `criterion` (as well as `iai`) in the past and was planning to propose it as an improvement to the current benchmark suite in a separate change, as it is slightly more flexible and has nicer output. I think it might also allow us to do this, although I'm not sure about that.